VotingApplication/VotingApplication.Web.Api.Tests/E2E/BasicPollTests.cs

```diff
@@ -0,0 +1,603 @@
+                new Option(){ Name = "Test Option 2", 
+                              Description = "A very long test description 2 that should exceed the character limit for descriptions" }};
+
+                // Open, Anonymous, No Option Adding, Shown Results
+                _defaultBasicPoll = new Poll()
+                {
+                    UUID = Guid.NewGuid(),
+                    PollType = PollType.Basic,
+                    Name = "Test Poll",
+                    LastUpdated = DateTime.Now,
+                    CreatedDate = DateTime.Now,
+                    Options = testPollOptions,
+                    InviteOnly = false,
+                    NamedVoting = false,
+                    OptionAdding = false,
+                    HiddenResults = false
+                };
+
+                _context.Polls.Add(_defaultBasicPoll);
+                _context.SaveChanges();
+            }
+
+            [ClassCleanup]
+            public static void ClassCleanup()
+            {
+                PollClearer pollTearDown = new PollClearer(_context);
+                pollTearDown.ClearPoll(_defaultBasicPoll);
+                _context.Dispose();
+            }
+
+            [TestMethod, TestCategory("E2E")]
+            public void BasicVote_PopulatedOptions_DisplaysAllOptions()
+            {
+                _driver.Navigate().GoToUrl(SiteBaseUri + "Poll/#/Vote/" + _defaultBasicPoll.UUID);
+                IReadOnlyCollection<IWebElement> optionNames = _driver.FindElements(NgBy.Binding("option.Name"));
+
+                Assert.AreEqual(_defaultBasicPoll.Options.Count, optionNames.Count);
+                CollectionAssert.AreEquivalent(_defaultBasicPoll.Options.Select(o => o.Name).ToList(),
+                                               optionNames.Select(o => o.Text).ToList());
+            }
+
+            [TestMethod, TestCategory("E2E")]
+            public void BasicVote_PopulatedOptions_DisplaysAllOptionDescriptions()
+            {
+                _driver.Navigate().GoToUrl(SiteBaseUri + "Poll/#/Vote/" + _defaultBasicPoll.UUID);
+                IReadOnlyCollection<IWebElement> optionDescriptions = _driver.FindElements(NgBy.Model("option.Description"));
+
+                Assert.AreEqual(_defaultBasicPoll.Options.Count, optionDescriptions.Count);
+                List<String> expectedDescriptions = new List<string>();
+
+                foreach (Option option in _defaultBasicPoll.Options)
+                {
+                    string truncatedDescription = option.Description
+                                                          .Split(' ')
+                                                          .Aggregate((prev, curr) => (curr.Length + prev.Length >= truncatedTextLimit) ?
+                                                                                     prev : prev + " " + curr);
+
+                    if (truncatedDescription != option.Description)
```

Comment:
Danger, Will Robinson!
![danger-will-robinson](https://cloud.githubusercontent.com/assets/11292375/7488734/23047266-f3c6-11e4-871c-4c1d0b2fb97e.gif)

There's a fair bit of logic in this test. `if` statements in tests are especially a bad smell. This test feels like it's being shoehorned into using the common setup from previous tests, rather than making use of them if it can.